### ISSUE TITLE
Evidence Inference 2.0 released code

### ISSUE THREAD

**Issue by karlferliksson** [jayded/evidence-inference#14]
Hi,
Thanks for some interesting work! In the new paper Evidence Inference 2.0, you mention that code for the new baseline models (BERT-to-BERT pipeline) will be released. I can't seem to find it in the repo, do you know when it will be made available?
Thanks

**Comment by jayded** [comment]
Hi! I just pushed some WIP code to a [branch](https://github.com/jayded/evidence-inference/tree/ei2.0prerelease). We want to get a proper release out but I don't think the project is yet in that state.

Caveats:
- We offer both a plaintext and an nxml version of the data. Offsets into the nxml are broken, and will eventually be fixed and released. This is a main blocker.
- The models as I pushed them are set up to use Biomed RoBERTa. This had a weird behavior where it almost doubled the amount of memory the model used - the SciBERT version of the code ([tagged here](https://github.com/jayded/evidence-inference/releases/tag/v2.0prerelease); although it seems I missed the parameters - they're largely the same except for changing any mentions of `biomed_roberta_base` in [this file](https://github.com/jayded/evidence-inference/blob/ei2.0prerelease/params/bert_pipeline_ev2.0.json) and friends should probably be changed to something like `scibert_scivocab_uncased`). The SciBERT version ought to run on a 1080TI or anything with 11G of GPU memory; BiomedRoBERTa required a Titan for unknown reasons (try an updated transformers)
- The code is a little ugly. You can easily (but slowly) reproduce our experiments [here](https://github.com/jayded/evidence-inference/blob/ei2.0prerelease/evidence_inference/models/pipeline.py) but there are many things you probably want to turn off. You turn them off by commenting things out.
- I hope to clean that up and release an easier to use version, but that's targeted for closer to bionlp.
- Documentation is a little lacking
- I need to update the `requirements.txt` to be sure all the experiment configurations run happily, as well as update the instructions (e.g. I think I'm missing something about installing the [scispacy tokenizer](https://github.com/allenai/scispacy/blob/master/README.md) )
- We need to update the citation!

**Comment by karlferliksson** [comment]
Thanks for the quick and thorough reply, as well as the WIP code. I'll have a look at the new branch and see if I can get it running. Also looking forward to the proper release when it is ready.

**Comment by jayded** [comment]
Happy to help/generally answer questions. The work is only useful if, well, actually gets use.

**Comment by karlerliksson** [comment]
Hi again. After adding a few packages to `requirements.txt`, I managed to run parts of the pipeline script with BiomedRoBERTa. The classification scores are slightly worse than in Table 1 in the paper, but quite close. Thanks for your help.

**Comment by jayded** [comment]
Good to hear! I ran all the RoBERTa configurations exactly once, so you're seeing some natural variance out of the model. This is probably due to the optimizer parameters and training data ordering and sampling not being deterministic. I would have expected the results to bounce around the scores in Table 1; if they're uniformly lower that's a little surprising. 

We didn't run a more thorough analysis of how errors compound in the pipeline model although I suppose you could train a few of these and actually measure that if it would be of value to you. You'd want to change the default target to the validation set and probably turn off most of the pipeline stages as well so it finishes quickly.

**Comment by jayded** [comment]
@karlpeltarion this is a very late notification that the updated code has been released.
